azkaban-common/src/test/java/azkaban/project/ProjectManagerTest.java

```diff
@@ -0,0 +1,162 @@
+package azkaban.project;
+
+import azkaban.user.User;
+import azkaban.utils.Props;
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import org.apache.commons.io.FileUtils;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+public class ProjectManagerTest {
+
+  private File workingDir;
+
+  @Before
+  public void setUp() throws Exception {
+    System.out.println("projectManagerTest: create temp dir");
+    synchronized (this) {
+      workingDir = new File("_AzkabanTestDir_" + System.currentTimeMillis());
+      if (workingDir.exists()) {
+        FileUtils.deleteDirectory(workingDir);
+      }
+      workingDir.mkdirs();
+    }
+  }
+
+  @After
+  public void tearDown() throws IOException {
+    System.out.println("Teardown temp dir");
+    synchronized (this) {
+      if (workingDir != null) {
+        FileUtils.deleteDirectory(workingDir);
+        workingDir = null;
+      }
+    }
+  }
+
+  @Test
+  public void testCreateProject() throws ProjectManagerException {
+    System.out.println("projectManagerTest: testCreateProject");
+    ProjectManager manager = createProjectManager();
+    String projectName = "myTestCreateProject";
+    String projectDescription = "This is to test creating a project";
+    User user = new User("testUser");
+    createProjectUtil(manager, projectName, projectDescription, user);
+    System.out.println("Created project successfully");
+  }
+
+  @Test
+  public void testRemoveProject() throws ProjectManagerException {
+    System.out.println("projectManagerTest: testRemoveProject");
+    ProjectManager manager = createProjectManager();
+    String projectName = "myTestRemoveProject";
+    String projectDescription = "This is to test removing a project";
+    User user = new User("testUser");
+    Project project = createProjectUtil(manager, projectName, projectDescription, user);
+
+    manager.removeProject(project, user);
+    Project fetchedProject = manager.getProject(project.getId());
+    Assert.assertFalse(fetchedProject.isActive());
+    System.out.println("Removed project successfully");
+  }
+
+  @Test
+  public void testUploadProject() throws ProjectManagerException {
+    System.out.println("projectManagerTest: testUploadProject");
+    ProjectManager manager = createProjectManager();
+    String projectName = "myTestUploadProject";
+    String projectDescription = "This is to test uploading a project";
+    User user = new User("testUser");
+    Project project = createProjectUtil(manager, projectName, projectDescription, user);
+
+    File testDir = new File("src/test/resources/project/testjob/testjob.zip");
```

Comment:
Alternatively, you can consider this method:
http://stackoverflow.com/questions/2593154/get-a-resource-using-getresource